src/main/java/com/kainos/ea/controller/WebService.java

```diff
@@ -38,6 +61,125 @@ public Response getJobRoles()
 
     }
 
+    @GET
+    @Path("/job-roles/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getJobRoleById(@PathParam("id") int id){
+        try {
+            return Response.status(HttpStatus.OK_200).entity(rolesService.getRoleById(id)).build();
+        } catch (Exception | DatabaseConnectionException  e){
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500, e.getMessage()).build();
+        } catch (JobRoleDoesNotExistException e){
+            return Response.status(HttpStatus.NOT_FOUND_404, e.getMessage()).build();
+        }
+    }
+
+    @PUT
+    @Path("/job-roles")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response updateJobRole(JobRoleXL jobRoleXL){
```

Comment:
this should use the same model as the create job role endpoint

Reply:
changing JobRole to JobRoleResponse, JobRoleXL to JobRoleRequest